### PR TITLE
feat(api): add support for env switch by slug

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]

File: packages/shared/src/types/environment/index.ts
Comment (djabarovgeorge) [inline packages/shared/src/types/environment/index.ts:1]: extracted to shared

File: apps/dashboard/src/utils/types.ts [hunk 1]
```diff
@@ -1,10 +1,5 @@
 import type { StepResponseDto } from '@novu/shared';
 
-export enum BaseEnvironmentEnum {
-  DEVELOPMENT = 'Development',
-  PRODUCTION = 'Production',
-}
```
Comment (djabarovgeorge) [inline apps/dashboard/src/utils/types.ts:6]: duplication? 🧹

File: apps/web/src/studio/constants/EnvironmentEnum.ts
Comment (djabarovgeorge) [inline apps/web/src/studio/constants/EnvironmentEnum.ts:1]: moved to shared

File: apps/dashboard/src/components/create-workflow-button.tsx [hunk 2]
```diff
@@ -59,8 +59,8 @@ export const CreateWorkflowButton = (props: CreateWorkflowButtonProps) => {
       setIsOpen(false);
       navigate(
         buildRoute(ROUTES.EDIT_WORKFLOW, {
-          environmentId: currentEnvironment?._id ?? '',
-          workflowId: result.data._id,
+          environmentSlug: currentEnvironment?.slug ?? '',
+          workflowSlug: result.data.slug ?? '',
```
Comment (djabarovgeorge) [inline apps/dashboard/src/components/create-workflow-button.tsx:63]: bug fix

**Pull request merged**